src/modules/sdlog2/sdlog2.c

```diff
@@ -936,9 +936,12 @@ int sdlog2_thread_main(int argc, char *argv[])
 
 	flag_system_armed = false;
 
-	/* work around some stupidity in task_create's argv handling */
+#ifdef __PX4_NUTTX
```

Comment:
@julianoes Please don't remove workarounds without in-depth discussion and on-hardware cross checks.